cartoframes/dataset_info.py

```diff
@@ -0,0 +1,66 @@
+import time
+
+from carto.datasets import DatasetManager
+from carto.exceptions import CartoException
+
+
+class DatasetInfo():
+    PRIVATE = 'PRIVATE'
+    PUBLIC = 'PUBLIC'
+    LINK = 'LINK'
+
+    def __init__(self, carto_context, table_name):
+        self._get_metadata(carto_context, table_name)
+
+        if self._metadata is not None:
+            self.privacy = self._metadata.privacy
+            self.name = self._metadata.name
+        else:
+            raise CartoException('Something goes wrong accessing the table metadata.')
+
+    def update(self, privacy=None, name=None):
+        modified = False
+        if privacy and self._validate_privacy(privacy):
+            self.privacy = privacy.upper()
+            modified = True
+        if name and self._validate_name(name):
+            self.name = name
+            modified = True
+
+        if modified:
+            self._save_metadata()
+
+    def _get_metadata(self, carto_context, table_name, retries=6, retry_wait_time=1):
+        ds_manager = DatasetManager(carto_context.auth_client)
+        try:
+            self._metadata = ds_manager.get(table_name)
+        except Exception as e:
+            if type(e).__name__ == 'NotFoundException' and retries > 0:
+                # if retry_wait_time > 7: # it should be after more than 15 seconds
+                    # warn('We are still procesing the CARTO table. Sorry for the delay.')
+                time.sleep(retry_wait_time)
+                self._get_metadata(carto_context=carto_context, table_name=table_name,
+                                   retries=retries-1, retry_wait_time=retry_wait_time*2)
+            else:
+                return None
```

Comment:
Looks like it's a good practice, that the method returns the Exception when there's actually an exception instead of `None`, otherwise we lose the actual exception message and in some cases we'd need to add checks outside the method to see what happened.

Reply:
My idea here is not to expose a CARTO problem to the user in CARTOframes. I think the reason why we can not get the metadata it is not important for the user. But at the end, we are raising the error in the constructor https://github.com/CartoDB/cartoframes/pull/691/files/4cba183d2dbbc38deb56aa414ea11ae6c5bc3105#diff-43f2b42e1e2985dc2771f40dcf4c56dfR19 so we can move the raise from the constructor to here

Reply:
Here is where I miss the log part in CARTOframes hehe. I think we should provide here or in the constructor CARTO-information about the problem. Why? because if the user has a problem and she looks for help, we're not going to know how to help her because we don't know what is the problem. I'd raise the exception here, catch it in the constructor and provide meaningful information

Reply:
What is the meaningful information?  `We couldn't get the table metadata. Please, try again in a few seconds or contact support for help` as Alberto proposed before or the exception from pyrestcli?